microservices-dashboard-server/src/main/java/be/ordina/msdashboard/nodes/aggregators/health/HealthIndicatorsAggregator.java

```diff
@@ -61,78 +66,86 @@
+                .doOnCompleted(() -> logger.info("Completed merging all health observables"));
+    }
 
-	@Override
-	public Observable<Node> aggregateNodes() {
-		Observable<Observable<Node>> observableObservable = getServiceIdsFromDiscoveryClient()
-				.map(id -> new ImmutablePair<>(id, resolveHealthCheckUrl(id)))
-				.doOnNext(pair -> logger.info("Creating health observable: " + pair))
-				.map(pair -> getHealthNodesFromService(pair.getLeft(), pair.getRight()))
-				.doOnNext(el -> logger.debug("Unmerged health observable: " + el))
-				.doOnError(e -> errorHandler.handleSystemError("Error filtering services: " + e.getMessage(), e))
-				.doOnCompleted(() -> logger.info("Completed getting all health observables"))
-				.retry();
-		return Observable.merge(observableObservable)
-				.doOnNext(el -> logger.debug("Merged health node: " + el.getId()))
-				.doOnError(e -> errorHandler.handleSystemError("Error filtering services: " + e.getMessage(), e))
-				.doOnCompleted(() -> logger.info("Completed merging all health observables"));
-	}
+    private String resolveHealthCheckUrl(String id) {
+        List<ServiceInstance> instances = discoveryClient.getInstances(id);
+        if (instances.isEmpty()) {
+            throw new IllegalStateException("No instances found for service " + id);
+        } else {
+            return uriResolver.resolveHealthCheckUrl(instances.get(0));
+        }
+    }
 
-	private String resolveHealthCheckUrl(String id) {
-		List<ServiceInstance> instances = discoveryClient.getInstances(id);
-		if (instances.isEmpty()) {
-			throw new IllegalStateException("No instances found for service " + id);
-		} else {
-			return uriResolver.resolveHealthCheckUrl(instances.get(0));
-		}
-	}
+    protected Observable<String> getServiceIdsFromDiscoveryClient() {
+        logger.info("Discovering services for health");
+        return Observable.from(discoveryClient.getServices()).subscribeOn(Schedulers.io()).publish().autoConnect()
+                .map(id -> id.toLowerCase())
+                .filter(id -> !id.equals(ZUUL))
+                .doOnNext(s -> logger.debug("Service discovered: " + s))
+                .doOnError(e -> errorHandler.handleSystemError("Error filtering services: " + e.getMessage(), e))
+                .retry();
+    }
 
-	protected Observable<String> getServiceIdsFromDiscoveryClient() {
-		logger.info("Discovering services for health");
-		return Observable.from(discoveryClient.getServices()).subscribeOn(Schedulers.io()).publish().autoConnect()
-				.map(id -> id.toLowerCase())
-				.filter(id -> !id.equals(ZUUL))
-				.doOnNext(s -> logger.debug("Service discovered: " + s))
-				.doOnError(e -> errorHandler.handleSystemError("Error filtering services: " + e.getMessage(), e))
-				.retry();
-	}
+    protected Observable<Node> getHealthNodesFromService(String serviceId, String url, Authentication authentication) {
+        SecurityContextHolder.getContext().setAuthentication(authentication);
```

Comment:
Security is by doing this automatically on the classpath. Is this what we want?
Perhaps we can work around this by using a `@ConditionalOnClass` with ASM resolution of the class, which means we don't need to actually have it on the classpath. That way we could perhaps have an interface known to the application, with a default implementation and a security implementation. The latter would require the security dependency.